client/App/Containers/LoggingSamples/LoggingSamplesScreen.js

```diff
@@ -12,13 +12,28 @@ class LoggingSamplesScreen extends React.Component {
     this._fetchData();
   }
 
+  componentDidUpdate(prevProps){
+    if (this.props.sampleSent !== prevProps.sampleSent
+      && this.props.sampleSent){
+        this.setState({sampleCounter: this.state.sampleCounter+1});
```

Comment:
This will most likely work, but it is dangerous as setState is async.
Which means that this.state.sampleCounter won't always include what you think

----

I wrote this then went to the internet to search why it's so important
if this setState will happen twice one after antoher the sampleCounter might be increased by one just once instead of twice.
check out this [article](https://dmitripavlutin.com/react-hooks-stale-closures/#32-usestate). It uses hooks so ignore all the useState part but it shows the same issue.
soooooooo to sum up you should use: 
``` this.setState(prevState => ({sampleCounter: prevState.sampleCounter + 1})); ```

Hope you don't hate me right now, I'd love to explain to you more

Reply:
Hevanti :D

Reply:
God you are so smart I'm so jealous of your Tamar..

Reply:
האמת שזה קצת דפק את זה